### ISSUE TITLE
Installation fails although jupyter is installed for all versions of python I have installed

### ISSUE THREAD

**Issue by Jeklah** [wmvanvliet/jupyter-vim#27]

**Comment by wmvanvliet** [comment]
> JupyterConnect returns not an editor command.

This usually means that the initialization of Python and Jupyter inside Vim did not work, hence the code never got around to defining the JupyterConnect command. The key to solving this issue is to find out what exactly the error was that was generated by the initialization code. When starting Vim and opening a `.py` file, do you get any other messages flashing by?

**Comment by Jeklah** [comment]
I get an error about vim not being compiled with python but when i check the flags it has python3 and pythonx/dev

**Comment by wmvanvliet** [comment]
What does `: pythonx import sys; print(sys.path)` output?

**Comment by Jeklah** [comment]
`['/usr/lib/python3.7.zip', '/usr/lib/python3.7', '/usr/lib/python3.7/lib-dynload', '/home/Jeklah/.local/lib/python3.7/site-packages', '/usr/lib/python3.7/site-packages', '_vim_path_']`

I noticed when I run jupyter it runs with version 3.5 of Python..which I thought was a bit dated.

**Comment by wmvanvliet** [comment]
How do you get an error about vim not being compiled with python, when it clearly is?

**Comment by Jeklah** [comment]
> How do you get an error about vim not being compiled with python, when it clearly is?
This is an extremely good question and the root of my problems!
I think I made some slight progress this morning, I commented out some lines in my .vimrc.local and Pymode seems to be working now, so I can only assume vim is picking up python now, but I still getting JupyterConnect is not a valid editor command

**Comment by Jeklah** [comment]
I've managed to get jupyter using python 3.8 now instead of 3.5

**Comment by wmvanvliet** [comment]
The key thing to understand about this plugin is that *vim* also needs access to some version jupyter, so it can connect to a kernel. This means you need to have jupyter installed somewhere in the path that you obtained with your vim command above, for example `/usr/lib/python3.7/site-packages`. Check if `jupyter_client` is there. If not, use a command like `sudo /usr/bin/pip3 install jupyter` to install it.

**Comment by Jeklah** [comment]
I do have jupyter-client installed for all the versions of python I have on my machine, if I recall correctly.
I will double check this and post screenshots when I get home.
Thanks for your help, greatly appreciated. It'd be great if I could get this plugin working.

**Comment by Jeklah** [comment]
My apoligies for not checking last night, rl bit busy. Will definately verify this tonight first thing when I get home.

**Comment by Jeklah** [comment]
![jupyter-vim](https://user-images.githubusercontent.com/7274473/76640240-7ba95600-6547-11ea-9aed-c3eac42b7cea.png)

**Comment by Jeklah** [comment]
What are your thoughtts?

**Comment by wmvanvliet** [comment]
does this work in your vim?

`:pythonx import jupyter`

If that works, then that eliminates the most common source of failures for the plugin, and I need more information in order to help you.

**Comment by Jeklah** [comment]
Will try as soon as I get home.

**Comment by Jeklah** [comment]
Well it runs, no output, no error....so I can only it assume it ran successfully?

**Comment by wmvanvliet** [comment]
Yes, that means it ran successfully. I'm sorry, but from here on I would resort to print() debugging of the code. See what portions of the init function get run and where things go awry. Without any error messages to go on, that's the only thing I can think of.

**Comment by Jeklah** [comment]
Any idea where to start?
I'm not sure where to begin debugging this.

**Comment by wmvanvliet** [comment]
I would start here:
https://github.com/wmvanvliet/jupyter-vim/blob/master/autoload/jupyter.vim#L22-L37

**Comment by Jeklah** [comment]
Thank you, will investigate when I am home.

**Comment by Jeklah** [comment]
vim_path, _ = os.path.split(vim.eval("expand(''<sfile>:p:h'')"))

This line errors with :no source file name to substitute for sfile.

**Comment by Jeklah** [comment]
I will say I don't have experience with scripting vim but am a software engineer so can take educated guesses...my initial guess is that vim.eval isn't expanding what it should do, which it looks to me like it's cutting the python path from os.path.split...?
Writing this inbetween spare moments at work otherwise would do more indepth testing rather than typing out my thoughts.
Same as the other day, will investigate more once home..and with Covid 19, it's likely I'll be working from home soon so...maybe we'll solve this bug/misconfig problem yet!

Thanks again for your help so far.

**Comment by wmvanvliet** [comment]
Good start! I'm grateful for your documentation of this issue. You're not only solving this for yourself, but everyone that comes after you with the same problem :)

**Comment by Jeklah** [comment]
Thanks :)
I want to start contributing more to open source, so this is a good start I guess. :)

**Comment by karlmcguire** [comment]
I'm having this same issue. Using NeoVim, no errors, `:python3 import sys; print(sys.version)` and  `:python3 import jupyter` both run fine.
